docs/src/miden_library.md

```diff
@@ -0,0 +1,128 @@
+| `get_initial_storage_commitment` | Returns the storage commitment of the native account at the beginning of the transaction.<br><br>Inputs: `[]`<br>Outputs: `[INIT_STORAGE_COMMITMENT]`<br>Context: Any |
+| `compute_storage_commitment` | Computes the latest account storage commitment of the current account.<br><br>Inputs: `[]`<br>Outputs: `[STORAGE_COMMITMENT]`<br>Context: Account |
+| `get_balance` | Returns the balance of the fungible asset associated with the provided faucet_id in the current account's vault.<br><br>Inputs: `[faucet_id_prefix, faucet_id_suffix]`<br>Outputs: `[balance]`<br>Context: Any |
+| `has_non_fungible_asset` | Returns a boolean indicating whether the non-fungible asset is present in the current account's vault.<br><br>Inputs: `[ASSET]`<br>Outputs: `[has_asset]`<br>Context: Any |
+| `add_asset` | Adds the specified asset to the vault. For fungible assets, returns the total after addition.<br><br>Inputs: `[ASSET]`<br>Outputs: `[ASSET']`<br>Context: Native & Account |
+| `remove_asset` | Removes the specified asset from the vault.<br><br>Inputs: `[ASSET]`<br>Outputs: `[ASSET]`<br>Context: Native & Account |
+| `get_initial_vault_root` | Returns the vault root of the native account at the beginning of the transaction.<br><br>Inputs: `[]`<br>Outputs: `[INIT_VAULT_ROOT]`<br>Context: Any |
+| `get_vault_root` | Returns the vault root of the current account.<br><br>Inputs: `[]`<br>Outputs: `[VAULT_ROOT]`<br>Context: Any |
+| `was_procedure_called` | Checks if a procedure has been called during transaction execution.<br><br>Inputs: `[PROC_ROOT]`<br>Outputs: `[was_called]`<br>Context: Any |
+
+## Note Procedures (`miden::note`)
+
+Note procedures can be used to fetch data from the note that is currently being processed and manipulate note assets.
+
+| Procedure | Description |
+| --- | --- |
+| `get_assets` | Writes the assets of the currently executing note into memory starting at the specified address.<br><br>Inputs: `[dest_ptr]`<br>Outputs: `[num_assets, dest_ptr]`<br>Context: Note |
+| `get_inputs` | Loads the note's inputs to the specified memory address.<br><br>Inputs: `[dest_ptr]`<br>Outputs: `[num_inputs, dest_ptr]`<br>Context: Note |
+| `get_sender` | Returns the sender of the note currently being processed.<br><br>Inputs: `[]`<br>Outputs: `[sender_id_prefix, sender_id_suffix]`<br>Context: Note |
+| `get_serial_number` | Returns the serial number of the note currently being processed.<br><br>Inputs: `[]`<br>Outputs: `[SERIAL_NUMBER]`<br>Context: Note |
+| `get_script_root` | Returns the script root of the note currently being processed.<br><br>Inputs: `[]`<br>Outputs: `[SCRIPT_ROOT]`<br>Context: Note |
+| `compute_inputs_commitment` | Computes the commitment to the note inputs starting at the specified memory address.<br><br>Inputs: `[inputs_ptr, num_inputs]`<br>Outputs: `[COMMITMENT]`<br>Context: Local |
```

Comment:
Not for this PR, but wonder if this should go to a different module as this procedure is the only procedure in `miden::note` module which is not related to the currently executing note.